### ISSUE TITLE
Error: undefined constant TestValue

### ISSUE THREAD

**Issue by postmodern** [icy-arctic-fox/spectator#34]
Suddenly started getting this error with spectator ~> 0.9 and ~> 0.10.

```
$ crystal spec --error-trace
error in line 4
Error: while requiring "./spec/chars_spec.cr"


In spec/chars_spec.cr:4:11

 4 | Spectator.describe Chars do
               ^-------
Error: expanding macro


In spec/chars_spec.cr:4:1

 4 | Spectator.describe Chars do
     ^
Error: expanding macro


There was a problem expanding macro 'describe'

Called macro defined in macro 'macro_140000956373072'

 46 | macro describe(description, *tags, **metadata, &block)

Which expanded to:

 >   1 |         class ::SpectatorTestContext
 >   2 |           describe(Chars,  ) do
 >   3 |   describe("NUMERIC") do
 >   4 |     subject do
 >   5 |       Chars::NUMERIC
 >   6 |     end
 >   7 |     it("should provide a numeric CharSet") do
 >   8 |       (expect(subject)).to(be =~ "0123456789")
 >   9 |     end
 >  10 |   end
 >  11 |   describe("OCTAL") do
 >  12 |     subject do
 >  13 |       Chars::OCTAL
 >  14 |     end
 >  15 |     it("should provide an octal CharSet") do
 >  16 |       (expect(subject)).to(be =~ "01234567")
 >  17 |     end
 >  18 |   end
 >  19 |   describe("UPPERCASE_HEXADECIMAL") do
 >  20 |     subject do
 >  21 |       Chars::UPPERCASE_HEXADECIMAL
 >  22 |     end
 >  23 |     it("should provide an upper-case hexadecimal CharSet") do
 >  24 |       (expect(subject)).to(be =~ "0123456789ABCDEF")
 >  25 |     end
 >  26 |   end
 >  27 |   describe("LOWERCASE_HEXADECIMAL") do
 >  28 |     subject do
 >  29 |       Chars::LOWERCASE_HEXADECIMAL
 >  30 |     end
 >  31 |     it("should provide a lower-case hexadecimal CharSet") do
 >  32 |       (expect(subject)).to(be =~ "0123456789abcdef")
 >  33 |     end
 >  34 |   end
 >  35 |   describe("HEXADECIMAL") do
 >  36 |     subject do
 >  37 |       Chars::HEXADECIMAL
 >  38 |     end
 >  39 |     it("should provide a hexadecimal CharSet") do
 >  40 |       (expect(subject)).to(be =~ "0123456789ABCDEFabcdef")
 >  41 |     end
 >  42 |   end
 >  43 |   describe("UPPERCASE_ALPHA") do
 >  44 |     subject do
 >  45 |       Chars::UPPERCASE_ALPHA
 >  46 |     end
 >  47 |     it("should provide an upper-case alpha CharSet") do
 >  48 |       (expect(subject)).to(be =~ "ABCDEFGHIJKLMNOPQRSTUVWXYZ")
 >  49 |     end
 >  50 |   end
 >  51 |   describe("LOWERCASE_ALPHA") do
 >  52 |     subject do
 >  53 |       Chars::LOWERCASE_ALPHA
 >  54 |     end
 >  55 |     it("should provide a lower-case alpha CharSet") do
 >  56 |       (expect(subject)).to(be =~ "abcdefghijklmnopqrstuvwxyz")
 >  57 |     end
 >  58 |   end
 >  59 |   describe("ALPHA") do
 >  60 |     subject do
 >  61 |       Chars::ALPHA
 >  62 |     end
 >  63 |     it("should provide an alpha CharSet") do
 >  64 |       (expect(subject)).to(be =~ "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz")
 >  65 |     end
 >  66 |   end
 >  67 |   describe("ALPHA_NUMERIC") do
 >  68 |     subject do
 >  69 |       Chars::ALPHA_NUMERIC
 >  70 |     end
 >  71 |     it("should provide an alpha-numeric CharSet") do
 >  72 |       (expect(subject)).to(be =~ "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789")
 >  73 |     end
 >  74 |   end
 >  75 |   describe("VISIBLE") do
 >  76 |     subject do
 >  77 |       Chars::VISIBLE
 >  78 |     end
 >  79 |     it("should provide a visible CharSet") do
 >  80 |       (expect(subject)).to(be =~ "!\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~")
 >  81 |     end
 >  82 |   end
 >  83 |   describe("SPACE") do
 >  84 |     subject do
 >  85 |       Chars::SPACE
 >  86 |     end
 >  87 |     it("should provide a space CharSet") do
 >  88 |       (expect(subject)).to(be =~ "\t\n\v\f\r ")
 >  89 |     end
 >  90 |   end
 >  91 |   describe("PUNCTUATION") do
 >  92 |     subject do
 >  93 |       Chars::PUNCTUATION
 >  94 |     end
 >  95 |     it("should provide a punctuation CharSet") do
 >  96 |       (expect(subject)).to(be =~ " !\"'(),-.:;?[]`{}~")
 >  97 |     end
 >  98 |   end
 >  99 |   describe("SYMBOLS") do
 > 100 |     subject do
 > 101 |       Chars::SYMBOLS
 > 102 |     end
 > 103 |     it("should provide a symbols CharSet") do
 > 104 |       (expect(subject)).to(be =~ " !\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~")
 > 105 |     end
 > 106 |   end
 > 107 | end
 > 108 |         end
 > 109 |       
Error: expanding macro


In spec/chars_spec.cr:15:11

 15 | 
                ^-------
Error: expanding macro


In spec/chars_spec.cr:8:1

 8 | it "should provide a numeric CharSet" do
 ^-
Error: expanding macro


In spec/chars_spec.cr:8:1

 8 | it "should provide a numeric CharSet" do
 ^
Error: expanding macro


There was a problem expanding macro 'it'

Called macro defined in macro 'define_example'

 17 | macro it(what = nil, *tags, **metadata, &block)

Which expanded to:

 >  1 |         
 >  2 |         
 >  3 | 
 >  4 |         _spectator_metadata(__temp_36, :metadata,  )
 >  5 |         _spectator_metadata(__temp_860, __temp_36,  )
 >  6 | 
 >  7 |         
 >  8 |           
 >  9 | 
 > 10 |           private def __temp_861() : Nil
 > 11 |             (expect(subject)).to(be =~ "0123456789")
 > 12 |           end
 > 13 | 
 > 14 |           ::Spectator::DSL::Builder.add_example(
 > 15 |             _spectator_example_name("should provide a numeric CharSet"),
 > 16 |             ::Spectator::Location.new("/data/home/postmodern/code/crystal/chars.cr/spec/chars_spec.cr", 8, 10),
 > 17 |             -> { new.as(::Spectator::Context) },
 > 18 |             __temp_860
 > 19 |           ) do |example|
 > 20 |             example.with_context(SpectatorTestContext::Group__temp_852::Group__temp_856) do
 > 21 |               
 > 22 |                 __temp_861
 > 23 |               
 > 24 |             end
 > 25 |           end
 > 26 | 
 > 27 |         
 > 28 |       
Error: instantiating 'Spectator::Example#with_context(SpectatorTestContext::Group__temp_852::Group__temp_856.class)'


In spec/chars_spec.cr:8:1

 8 | it "should provide a numeric CharSet" do
 ^
Error: expanding macro


There was a problem expanding macro 'it'

Called macro defined in macro 'define_example'

 17 | macro it(what = nil, *tags, **metadata, &block)

Which expanded to:

 >  1 |         
 >  2 |         
 >  3 | 
 >  4 |         _spectator_metadata(__temp_36, :metadata,  )
 >  5 |         _spectator_metadata(__temp_860, __temp_36,  )
 >  6 | 
 >  7 |         
 >  8 |           
 >  9 | 
 > 10 |           private def __temp_861() : Nil
 > 11 |             (expect(subject)).to(be =~ "0123456789")
 > 12 |           end
 > 13 | 
 > 14 |           ::Spectator::DSL::Builder.add_example(
 > 15 |             _spectator_example_name("should provide a numeric CharSet"),
 > 16 |             ::Spectator::Location.new("/data/home/postmodern/code/crystal/chars.cr/spec/chars_spec.cr", 8, 10),
 > 17 |             -> { new.as(::Spectator::Context) },
 > 18 |             __temp_860
 > 19 |           ) do |example|
 > 20 |             example.with_context(SpectatorTestContext::Group__temp_852::Group__temp_856) do
 > 21 |               
 > 22 |                 __temp_861
 > 23 |               
 > 24 |             end
 > 25 |           end
 > 26 | 
 > 27 |         
 > 28 |       
Error: instantiating 'Spectator::Example#with_context(SpectatorTestContext::Group__temp_852::Group__temp_856.class)'


In spec/chars_spec.cr:8:1

 8 | it "should provide a numeric CharSet" do
 ^
Error: expanding macro


There was a problem expanding macro 'it'

Called macro defined in macro 'define_example'

 17 | macro it(what = nil, *tags, **metadata, &block)

Which expanded to:

 >  1 |         
 >  2 |         
 >  3 | 
 >  4 |         _spectator_metadata(__temp_36, :metadata,  )
 >  5 |         _spectator_metadata(__temp_860, __temp_36,  )
 >  6 | 
 >  7 |         
 >  8 |           
 >  9 | 
 > 10 |           private def __temp_861() : Nil
 > 11 |             (expect(subject)).to(be =~ "0123456789")
 > 12 |           end
 > 13 | 
 > 14 |           ::Spectator::DSL::Builder.add_example(
 > 15 |             _spectator_example_name("should provide a numeric CharSet"),
 > 16 |             ::Spectator::Location.new("/data/home/postmodern/code/crystal/chars.cr/spec/chars_spec.cr", 8, 10),
 > 17 |             -> { new.as(::Spectator::Context) },
 > 18 |             __temp_860
 > 19 |           ) do |example|
 > 20 |             example.with_context(SpectatorTestContext::Group__temp_852::Group__temp_856) do
 > 21 |               
 > 22 |                 __temp_861
 > 23 |               
 > 24 |             end
 > 25 |           end
 > 26 | 
 > 27 |         
 > 28 |       
Error: instantiating '__temp_861()'


In spec/chars_spec.cr:9:29

 9 | expect(subject).to be =~ "0123456789"
                           ^-
Error: instantiating 'Spectator::Matchers::TruthyMatcher#=~(String)'


In lib/spectator/src/spectator/matchers/truthy_matcher.cr:101:18

 101 | expected = TestValue.new(value)
                  ^--------
Error: undefined constant TestValue
```

**Comment by icy-arctic-fox** [comment]
Should be fixed on master, can you try it? The following snippet works:

```crystal
Spectator.describe "GitHub issue #34" do
  specify do
    expect(/foo/).to be === "foo"
    expect(/foo/).to be =~ "foo"
  end
end
```

**Comment by postmodern** [comment]
Works now. Thanks!

**Comment by postmodern** [comment]
When could we possibly get a new release for this fix? My other crystal shards are failing due to this bug in 0.10.1.

**Comment by icy-arctic-fox** [comment]
Sorry about the wait. I'll cut a release shortly.

**Comment by icy-arctic-fox** [comment]
Released and officially fixed in v0.10.2
